File: Java/org/oobench/patterns/mvc/ColorsModel.java
package org.oobench.patterns.mvc;

import org.oobench.patterns.mvc.common.*;
import java.util.*;

public class ColorsModel extends Model {
    private Map colors = new HashMap();

    public String rgbForColor(String color) {
        return (String)colors.get(color);
    }

    public Map getColors() {
        return colors;
    }

    public void addColorWithName(String name, String rgbValue) {
        colors.put(name, rgbValue);
        notifyObservers();
    }
}

File: C++/Patterns/MVC/ColorsModel.cpp
#include "MVC.hpp"
#include "ColorsModel.hpp"

namespace oobench {

    std::string& ColorsModel::getRGBForColor(std::string& color) {
        return colors[color];
    }

    std::map<std::string, std::string> ColorsModel::getColors() const {
        return colors;
    }
}
